### PR TITLE
Implement AddImuDataHandlerTest

### PR DESCRIPTION
**Pull request by cschuet**
This PR introduces first gRPC handler test that takes advantage of new RpcHandlerTestServer.

### CODE REVIEW

File: cartographer_grpc/handlers/add_imu_data_handler_test.cc [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+#include "cartographer_grpc/testing/mock_map_builder_context.h"
+#include "google/protobuf/text_format.h"
+#include "google/protobuf/util/message_differencer.h"
+#include "gtest/gtest.h"
+
+using ::testing::_;
+using ::testing::Eq;
+using ::testing::Pointee;
+using ::testing::Return;
+using ::testing::Test;
```
Comment (pifon2a) [inline cartographer_grpc/handlers/add_imu_data_handler_test.cc]: nit: move "using ..." into namespace { below.

File: cartographer_grpc/handlers/add_imu_data_handler_test.cc [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+using ::testing::Truly;
+
+namespace cartographer_grpc {
+namespace handlers {
+namespace {
+
+const std::string kMessage = R"PROTO(
+sensor_metadata {
+  trajectory_id: 1
```
Comment (pifon2a) [inline cartographer_grpc/handlers/add_imu_data_handler_test.cc]: nit: indent everything inside the proto raw string.

File: cartographer_grpc/handlers/add_imu_data_handler_test.cc [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+ protected:
+  std::unique_ptr<framework::testing::RpcHandlerTestServer<AddImuDataHandler>>
+      test_server_;
+  testing::MockMapBuilderContext *mock_map_builder_context_;
+  std::unique_ptr<testing::MockLocalTrajectoryUploader>
+      mock_local_trajectory_uploader_;
+  proto::AddImuDataRequest request_;
+};
+
+DataPredicateType DataPredicate(const proto::AddImuDataRequest &proto) {
```
Comment (gaschler) [inline cartographer_grpc/handlers/add_imu_data_handler_test.cc]: rename to BuildPredicateDataEquals?

File: cartographer_grpc/framework/server.h [hunk 4]
```diff
@@ -80,14 +80,13 @@ class Server {
                       typename RpcHandlerType::OutgoingType>::value,
               method_full_name});
     }
+    static std::tuple<std::string /* service_full_name */,
+                      std::string /* method_name */>
+    ParseMethodFullName(const std::string& method_full_name);
```
[the remark this answers is not in the document]
Comment (cschuet) [reply cartographer_grpc/framework/server.h]: Needs to be public.

File: cartographer_grpc/framework/testing/rpc_handler_test_server.h [hunk 5]
```diff
@@ -21,131 +21,94 @@
...
+                                     grpc::InsecureChannelCredentials())),
+        client_(channel_) {
+    std::string method_full_name =
+        RpcHandlerInterface::Instantiate<RpcHandlerType>()->method_name();
+    std::string service_full_name;
+    std::string method_name;
+    std::tie(service_full_name, method_name) =
+        Server::Builder::ParseMethodFullName(method_full_name);
+
+    // Register the handler under test and set the ExecutionContext.
```
[the remark this answers is not in the document]
Comment (cschuet) [reply cartographer_grpc/framework/testing/rpc_handler_test_server.h]: Done.

File: cartographer_grpc/handlers/add_imu_data_handler_test.cc (hunk 1 above)
Comment (cschuet) [reply cartographer_grpc/handlers/add_imu_data_handler_test.cc]: done.

File: cartographer_grpc/handlers/add_imu_data_handler_test.cc (hunk 2 above)
Comment (cschuet) [reply cartographer_grpc/handlers/add_imu_data_handler_test.cc]: Done.

File: cartographer_grpc/handlers/add_imu_data_handler_test.cc (hunk 3 above)
Comment (cschuet) [reply cartographer_grpc/handlers/add_imu_data_handler_test.cc]: done

Comment (cschuet) [reply cartographer_grpc/handlers/add_imu_data_handler_test.cc]: done.

File: cartographer_grpc/testing/mock_local_trajectory_uploader.h [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+#ifndef CARTOGRAPHER_GRPC_TESTING_MOCK_LOCAL_TRAJECTORY_UPLOADER_H
+#define CARTOGRAPHER_GRPC_TESTING_MOCK_LOCAL_TRAJECTORY_UPLOADER_H
+
+#include "cartographer_grpc/local_trajectory_uploader.h"
+#include "glog/logging.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::_;
```
[the remark this answers is not in the document]
Comment (cschuet) [reply cartographer_grpc/testing/mock_local_trajectory_uploader.h]: done.

File: cartographer_grpc/testing/mock_map_builder_context.h [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+#ifndef CARTOGRAPHER_GRPC_TESTING_MOCK_MAP_BUILDER_CONTEXT_H
+#define CARTOGRAPHER_GRPC_TESTING_MOCK_MAP_BUILDER_CONTEXT_H
+
+#include "cartographer/mapping/local_slam_result_data.h"
+#include "cartographer_grpc/map_builder_context_interface.h"
+#include "glog/logging.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::_;
```
[the remark this answers is not in the document]
Comment (cschuet) [reply cartographer_grpc/testing/mock_map_builder_context.h]: done.

**Pull request merged**